packages/xo-server/src/xo-mixins/metadata-backups.js

```diff
@@ -347,22 +349,32 @@ export default class metadataBackup {
 
         let outputStream
         try {
-          await Promise.all([
+          const backupResolution = await allSettled([
             (async () => {
               outputStream = await handler.createOutputStream(fileName)
 
               // 'readable-stream/pipeline' not call the callback when an error throws
               // from the readable stream
               stream.pipe(outputStream)
-              return fromEvent(stream, 'end').catch(error => {
-                if (error.message !== 'aborted') {
-                  throw error
-                }
-              })
+              return timeout.call(
+                fromEvent(stream, 'end').catch(error => {
+                  if (error.message !== 'aborted') {
+                    throw error
+                  }
+                }),
+                this._poolMetadataTimeout
+              )
             })(),
             handler.outputFile(metaDataFileName, metadata),
           ])
 
+          const firstRejectedPromise = backupResolution.find(
+            _ => _.status === ALL_SETTLED_REJECTED
+          )
+          if (firstRejectedPromise !== undefined) {
+            throw firstRejectedPromise.reason
+          }
```

Comment:
This looks like a common use case, can we make it easier?

Reply:
Do you mean implement it in the `allSettled` module?

Reply:
Why not.